### PR TITLE
feat(flow): add flow types to dist

### PR DESCRIPTION
**Pull request by arahansen**
- adds build step to copy src files and output to dist as `*.flow`. These files will be picked up by `flow` when type checking
- fixed an issue with flow config where flow was looking at dist for `reflex` components. Although, this could theoretically be resolved by the new build step implemented here.

Resolves https://github.com/obartra/reflex/issues/144

### CODE REVIEW

**Comment by arahansen** [comment]
Unfortunately not from what I can tell. :(
https://flow.org/en/docs/types/modules/#importing-and-exporting-types-a-classtoc-idtoc-importing-and-exporting-types-hreftoc-importing-and-exporting-typesa

File: storybook/stories/components/search/nav.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { Col, Layout, Grid, Root } from 'reflex'
 
 export default function SearchNav() {
   return (
-    <Layout dev={4} margin={[1, 0, 0, 0]}>
+    <Layout dev={4} margin={[3, 1, 4, 0]}>
```
Comment (arahansen) [inline storybook/stories/components/search/nav.js]: yep, should be reverted now

**Comment by obartra** [comment]
Yeah I was just looking at that. oh well don't worry about it. I do see some linting issues on flow types, maybe we should exclude that folder from linting on bithound/cc/eslint?

**Comment by arahansen** [comment]
Linting issues should be resolved.

**Pull request merged**